substrate/pkg/controller/substrate/cluster/config.go

```diff
@@ -369,22 +379,29 @@ type DirectoryIterator struct {
 }
 
 // NewDirectoryIterator builds a new DirectoryIterator
-func NewDirectoryIterator(bucket, dir string) s3manager.BatchUploadIterator {
+func NewDirectoryIterator(bucket, dir string) (s3manager.BatchUploadIterator, error) {
 	var paths []string
-	filepath.Walk(dir, func(path string, info os.FileInfo, err error) error {
-		if err != nil {
-			return err
-		}
-		if !info.IsDir() {
-			paths = append(paths, path)
-		}
-		return nil
-	})
+	err := filepath.Walk(
```

Comment:
> Breaking args in multiple lines is not the ideal way, it makes reading harder since you need to now traverse multiple lines to understand the complete flow. I strongly suggest keeping them in a single line as the line length in not too long.

I think it makes sense to break it into multiple lines in this case, as one of the parameters for `filepath.Walk` is a function definition, it is hard to keep this function definition in one line.

One approach I can do is, to define and name this function, and use the function variable instead of its full definition as one of the parameters, and keep the `filepath.Walk` function call in one line.